components/raftstore/src/store/fsm/peer.rs

```diff
@@ -2393,31 +2416,113 @@ where
         }
     }
 
+    /// Check if destroy can be executed immediately. If it can't, the reason is returned.
+    fn maybe_delay_destroy(&mut self) -> Option<DelayReason> {
+        if self.fsm.peer.has_unpersisted_ready() {
+            assert!(self.ctx.sync_write_worker.is_none());
+            // The destroy must be delayed if there are some unpersisted readies.
+            // Otherwise there is a race of writting kv db and raft db between here
+            // and write worker.
+            return Some(DelayReason::UnPersistedReady);
+        }
+
+        if !self.fsm.logs_gc_flushed {
+            let start_index = self.fsm.peer.last_compacted_idx;
+            let mut end_index = start_index;
```

Comment:
if `start_index` == `end_index`, the task is only to make sure all the previous gc tasks of the region are flushed. Can we set the `end_index` with `last_index`, so there is nearly nothing to write in the following destroy process of raftstore.

Reply:
It may be okay if we also let the log gc worker write other metadata (like `PeerState`, `RegionLocalState`) in the same `WriteBacth` that deletes raft log

Reply:
I don't think it's a good idea to make these essential operations happen in multiple places.